File: cap08/src/test/java/app/car/cap08/infrastructure/FileUtils.java
package app.car.cap08.infrastructure;

import lombok.SneakyThrows;
import org.springframework.core.io.ClassPathResource;

import java.io.InputStream;
import java.util.Map;

public class FileUtils {


    @SneakyThrows
    public static String loadFileContents(String fileName) {

        InputStream is = new ClassPathResource(fileName).getInputStream();
        byte[] data = new byte[is.available()];
        is.read(data);
        return new String(data);
    }

    public static String loadFileContents(String fileName, Map<String, String> replacements) {
        String fileContents = loadFileContents(fileName);

        for (Map.Entry<String, String> entry : replacements.entrySet()) {
            fileContents = fileContents.replace("{{" + entry.getKey() + "}}", entry.getValue());
        }
        return fileContents;
    }
}

File: cap09/src/test/kotlin/app/car/cap09/infrastructure/FileUtils.kt
package app.car.cap09.infrastructure

import org.springframework.core.io.ClassPathResource

fun loadFileContents(fileName: String, replacements: Map<String, String> = emptyMap()) : String {
    val fileContents = ClassPathResource(fileName).inputStream.readAllBytes().decodeToString()
    fileContents.apply {
        var contents = this
        replacements.forEach { (key, value) -> contents = contents.replace("{{$key}}", value) }
        return contents
    }
}
